Automattic/sensei
# Show featured course label on course list block

Fixes #5405

### Changes proposed in this Pull Request
- Add Course Featured block 
- This block doesn't have content but we use it to get the information if the course is featured
- We use **render_block** [filter](https://developer.wordpress.org/reference/functions/render_block/)  to manipulate the content on the page render for preview
- in the edit, we hook onto EditBlock to trigger the same behavior of adding a Course Featured Badge and the wrapper around. 

### Notes; 
1. We have the playwright test failing, I would first want to hear the feedback before diving into fixing them  

### Testing instructions
- Add 4 courses 
- Add a featured image to 2 courses
- Add categories to one course with an image
- Add categories to one course without an image
- Mark course featured to 3 of these courses
- Make sure it looks as expected in all the cases for both Preview and Edit
- Make sure it looks good with different patterns and different themes

<img width="471" alt="image" src="https://user-images.githubusercontent.com/7208249/188462852-2463d034-24b1-43de-b624-00348c2735ad.png">
<img width="586" alt="image" src="https://user-images.githubusercontent.com/7208249/188462880-b984c27e-1e6f-486b-ab59-9d78cc3bbd35.png">
